library/multipicker/src/main/java/im/vector/lib/multipicker/utils/ImageUtils.kt

```diff
@@ -16,23 +16,28 @@
 
 package im.vector.lib.multipicker.utils
 
+import android.content.ContentResolver
 import android.content.Context
 import android.graphics.Bitmap
 import android.graphics.BitmapFactory
 import android.graphics.ImageDecoder
 import android.net.Uri
 import android.os.Build
 import androidx.exifinterface.media.ExifInterface
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
 import timber.log.Timber
 
 object ImageUtils {
 
-    fun getBitmap(context: Context, uri: Uri): Bitmap? {
+    suspend fun getBitmap(context: Context, uri: Uri): Bitmap? {
         return try {
             if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.P) {
-                ImageDecoder.decodeBitmap(ImageDecoder.createSource(context.contentResolver, uri))
+                runCatching {
+                    ImageDecoder.decodeBitmap(ImageDecoder.createSource(context.contentResolver, uri))
+                }.getOrThrow()
```

Comment:
Without this, we get the warning `Inappropriate blocking method call` which is caused because the method in question blocks the current thread and doesn't allow the coroutine to suspend and has the potential to throw. Adding this block makes our intent more explicit thus removing the warning

Reply:
I do not have the warning on my side, strange. Thanks for the update.